platform/configurator/configurator-api/src/main/java/org/codice/ddf/admin/configurator/Configurator.java

```diff
@@ -0,0 +1,195 @@
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String updatePropertyFile(Path propFile, Map<String, String> properties, boolean keepIgnored);
+
+    /**
+     * Gets the current key:value pairs set in the given property file.
+     *
+     * @param propFile the property file to query
+     * @return the current set of key:value pairs
+     */
+    Map<String, String> getProperties(Path propFile);
+
+    /**
+     * Updates a bundle configuration file in the system with the given set of new key:value pairs.
+     *
+     * @param configPid   the configId of the bundle configuration file to update
+     * @param configs     the set of key:value pairs to save in the configuration
+     * @param keepIgnored if true, then any keys already in the config file will retain their
+     *                    initial values if they are excluded from the {@code properties} param; if
+     *                    false, then the only config entires that will be in the updated file are those
+     *                    provided by the {@code properties} param
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String updateConfigFile(String configPid, Map<String, Object> configs, boolean keepIgnored);
+
+    /**
+     * Gets the current key:value pairs set in the given configuration file.
+     *
+     * @param configPid the configId of the bundle configuration file to query
+     * @return the current set of key:value pairs
+     */
+    Map<String, Object> getConfig(String configPid);
+
+    /**
+     * Creates a new managed service for the given factory.
+     *
+     * @param factoryPid the factoryPid of the service to create
+     * @param configs    the set of key:value pairs to save in the new managed service's configuration
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String createManagedService(String factoryPid, Map<String, Object> configs);
+
+    /**
+     * Deletes a managed service.
+     *
+     * @param configPid the configPid of the instance of the service to delete
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String deleteManagedService(String configPid);
+
+    Map<String, Map<String, Object>> getManagedServiceConfigs(String factoryPid);
+
+    /**
+     * Retrieves the service reference. The reference should only be used for reading purposes,
+     * any changes should be done through a commit
+     *
+     * @param serviceClass - Class of service to retrieve
+     * @return first found service reference of serviceClass
+     * @throws ConfiguratorException if any errors occur
+     */
+    <S> S getServiceReference(Class<S> serviceClass) throws ConfiguratorException;
```

Comment:
Ah. Actually, the only thing that should be passed in _is_ the interface's class, not any of its implementors. The service interface behaves similarly to a key for a lookup.